src/main/java/org/spongepowered/common/data/processor/data/entity/IgniteableDataProcessor.java

```diff
@@ -0,0 +1,157 @@
+        if (this.supports(dataHolder)) {
+            try {
+                final int oldFireDamageDelay = manipulator.fireDelay().get();
+                final int oldFireTicks = manipulator.fireTicks().get();
+                final IgniteableData merged =
+                        checkNotNull(function, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+                manipulator.set(Keys.FIRE_DAMAGE_DELAY, merged.fireDelay().get());
+                manipulator.set(Keys.FIRE_TICKS, merged.fireTicks().get());
+                final int fireDamageDelay = merged.fireDelay().get();
+                final int fireTicks = merged.fireTicks().get();
+                net.minecraft.entity.Entity entity = (net.minecraft.entity.Entity) dataHolder;
+                entity.fire = fireTicks;
+                entity.fireResistance = fireDamageDelay;
+                Collection<ImmutableValue<?>> successful = new ArrayList<ImmutableValue<?>>();
+                Collection<ImmutableValue<?>> replaced = new ArrayList<ImmutableValue<?>>();
+                successful.add(new ImmutableSpongeBoundedValue<Integer>(Keys.FIRE_TICKS, oldFireTicks, 0, intComparator(), 0, 999999999));
+                replaced.add(new ImmutableSpongeBoundedValue<Integer>(Keys.FIRE_TICKS, fireTicks, 0, intComparator(), 0, 999999999));
+                successful
+                        .add(new ImmutableSpongeBoundedValue<Integer>(Keys.FIRE_DAMAGE_DELAY, oldFireDamageDelay, 20, intComparator(), 0, 999999999));
+                replaced.add(new ImmutableSpongeBoundedValue<Integer>(Keys.FIRE_DAMAGE_DELAY, fireDamageDelay, 20, intComparator(), 0, 999999999));
+                return DataTransactionBuilder.successReplaceResult(successful, replaced);
+            } catch (Exception e) {
+                return DataTransactionBuilder.failResult(manipulator.getValues());
+            }
+
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public Optional<ImmutableIgniteableData> with(Key<? extends BaseValue<?>> key, Object value, ImmutableIgniteableData immutable) {
+        if (key.equals(Keys.FIRE_DAMAGE_DELAY)) {
+            return Optional.<ImmutableIgniteableData>of(new ImmutableSpongeIgniteableData(immutable.get(Keys.FIRE_TICKS).get(), (Integer) value));
+        } else if (key.equals(Keys.FIRE_TICKS)) {
+            return Optional.<ImmutableIgniteableData>of(new ImmutableSpongeIgniteableData(immutable.get(Keys.FIRE_DAMAGE_DELAY).get(),
+                    (Integer) value));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            return DataTransactionBuilder.failNoData();
+        } else {
+            return DataTransactionBuilder.builder().result(DataTransactionResult.Type.FAILURE).build();
```

Comment:
This is exactly the same as `DataTransactionBuilder.failNoData()` - use the former for both cases.